### PR TITLE
fix workflow execution for first run

### PR DESCRIPTION
**Pull request by sbcd90**
*Issue #, if available:*
https://github.com/opensearch-project/security-analytics/issues/606

*Description of changes:*

*CheckList:*
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/src/test/kotlin/org/opensearch/alerting/resthandler/WorkflowRestApiIT.kt [hunk 1]
```diff
@@ -1140,4 +1140,49 @@ class WorkflowRestApiIT : AlertingRestTestCase() {
...
+        assertTrue("incorrect version", createdVersion > 0)
+        assertEquals("Incorrect Location header", "$WORKFLOW_ALERTING_BASE_URI/$createdId", createResponse.getHeader("Location"))
+
+        val testDoc = """{
+            "message" : "This is an error from IAD region",
+            "test_field" : "us-west-2"
+        }"""
+
+        indexDoc(index, "1", testDoc)
+        Thread.sleep(80000)
```
Comment (eirsep) [inline alerting/src/test/kotlin/org/opensearch/alerting/resthandler/WorkflowRestApiIT.kt:1183]: NIT: adding 80 second sleeps makes tests flaky and long running we should try simulate 1 second interval jobs

can you change this to create workflow with periodicty of 1 second and sleep for 5 or 10 seconds?

**Review by eirsep — APPROVED** [review]

File: alerting/src/test/kotlin/org/opensearch/alerting/resthandler/WorkflowRestApiIT.kt (hunk 1 above)
Comment (sbcd90) [reply alerting/src/test/kotlin/org/opensearch/alerting/resthandler/WorkflowRestApiIT.kt:1183]: we cannot change it to `1 second` because the min frequency allowed is `1 min`. we can remove the test always or mark as ignored.

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt [hunk 2]
```diff
@@ -372,6 +381,42 @@ class TransportIndexWorkflowAction @Inject constructor(
...
+
+                    if (monitor.monitorType == Monitor.MonitorType.DOC_LEVEL_MONITOR) {
+                        docLevelMonitorQueries.indexDocLevelMonitorQueries(
+                            monitor,
+                            monitor.id,
+                            monitorMetadata,
+                            request.refreshPolicy,
+                            indexTimeout
+                        )
+                    }
```
Comment (lezzago) [inline alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt:416]: Why is this needed? This happens when the monitor is first created anyways.

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt [hunk 3]
```diff
@@ -499,6 +544,44 @@ class TransportIndexWorkflowAction @Inject constructor(
...
+                                .filter(QueryBuilders.matchQuery("monitor_id", monitor.id))
+                                .execute(it)
+                        }
+                        docLevelMonitorQueries.indexDocLevelMonitorQueries(
+                            monitor,
+                            monitor.id,
+                            updatedMetadata,
+                            request.refreshPolicy,
+                            indexTimeout
+                        )
```
Comment (lezzago) [inline alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt:581]: why do we need to reindex the monitor queries?

File: alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt [hunk 4]
```diff
@@ -3151,7 +3151,7 @@ class MonitorDataSourcesIT : AlertingSingleNodeTestCase() {
         } catch (ex: java.lang.Exception) {
             exception = ex
         }
-        assertTrue(exception is java.util.NoSuchElementException)
+        assertFalse(exception is java.util.NoSuchElementException)
```
Comment (lezzago) [inline alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt:3154]: Dont we want to move this into the catch and keep it as `assertTrue`?

Comment (sbcd90) [reply alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt]: fixed it.

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt (hunk 3 above)
Comment (sbcd90) [reply alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt]: fixed it.

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt (hunk 2 above)
Comment (sbcd90) [reply alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt]: fixed it.

**Review by lezzago — APPROVED** [review]
